doc/user-documentation/bb-to-std-mapping.md

```diff
@@ -0,0 +1,172 @@
+| `RelocatableBitSet` | - | no | yes | run-time fixed-size bitset from custom-allocated heap |
+| `Container` | `sync::channel`[^6] or `sync::sync_channel`[^7]  | yes | no | run-time fixed-sized mpmc FIFO queue from default heap |
+| `FixedSizeContainer` | `sync::channel`[^6] or `sync::sync_channel`[^7]   | yes | no | compile-time fixed-size mpmc FIFO queue map |
+| `RelocatableContainer` | `sync::channel`[^6] or `sync::sync_channel`[^7]   | no | yes | run-time fixed-size mpmc FIFO queue from custom-allocated heap |
+| `UniqueIndexSet` | `sync::channel`[^6] or `sync::sync_channel`[^7]  | no | yes | run-time fixed-sized mpmc FIFO queue for integers from custom-allocated heap |
+| `FixedSizeUniqueIndexSet` | `sync::channel`[^6] or `sync::sync_channel`[^7]   | yes | yes | compile-time version of `UniqueIndexSet` |
+
+[^6]: Currently still experimental in `1.85.0`. Unlike `std::sync::channel`, the `Container` is fixed-sized
+[^7]: Currently still experimental in `1.85.0`. Unlike `std::sync::sync_channel`, the `Container` is not blocking
+
+## POSIX abstraction (`iceoryx2_bb_posix`)
+
+### Concurrency
+
+| `iceoryx2` type | similar `std` type | short description |
+|-----------------|--------------------|---------|
+| `AdaptiveWait` | - | wait with auto-increasing waiting time to reduce CPU consumption  |
+| `Barrier` | `sync::Barrier` | |
+| `MultiConditionVariable` and `ConditionVariable` | `sync::CondVar` | block thread while waiting on multiple or single condition, respectively |
+| `DeadlineQueue` | - | wait on multiple periodic deadlines and allow monitoring the missed ones |
+| `MessageQueue` | - | async/sync channel with fixed-size queue size |
+| `Mutex` | `sync::Mutex` | mutual exclusion primitive |
+| `ReadWriteMutex` | `sync::RwLock` | allow multiple read-locks or one write-lock on shared resource |
+| `SharedMemory` | - | create, open and remove shared memory object |
+| `Signal` | - | POSIX signal handling |
+
+### Filesystem
+
+| `iceoryx2` type | similar `std` type | short description |
+|-----------------|--------------------|---------|
+| `Directory` and `DirectoryEntry` | `fs::DirEntry` | directory representation |
+| `File` | `fs::File` | read, create, write or modify files  |
+| `FileDescriptor` | `os::BorrowedFd` or `os::OwnedFd` | file descriptor representation |
+| `FileDescriptorSet` | - | useful for waiting on multiple objects |
+| `FileLock` | - | lock a file for exclusive writing or multiple reading |
+| `Group` | - | POSIX system's group representation |
+| `Metadata` | `fs::Metadata` | metadata info about a file (permissions, size, etc.) |
+| `Ownership` | - | file ownership representation (user and group) |
+| `Permission` | `fs::Permissions` | representation of various permissions on a file |
+
+### IPC
+
+| `iceoryx2` type | similar `std` type | short description |
+|-----------------|--------------------|---------|
+| `SharedMemory` | - | create, open and remove shared memory object |
```

Comment:
We also have an abstraction for a `NamedSemaphore` for IPC use cases, see: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/iceoryx2-bb/posix/src/semaphore.rs